### PR TITLE
Add timer utility and Cilkscale API example

### PR DESCRIPTION
**Pull request by ailiop**
This PR makes the following additions to bring the OpenCilk tutorials repo in sync with the [OpenCilk Getting Started
page](https://www.opencilk.org/doc/users-guide/getting-started/).

1. Add a simple C/C++ timer utility ([ctimer](https://github.com/ailiop/ctimer)).
2. Time the `fib()` execution and print the result.
3. Add example of using the Cilkscale API with `qsort` program.

Addresses issue [#172 (website repo)](https://github.com/OpenCilk/www.opencilk.org/issues/172).

### CODE REVIEW

**Comment by ailiop** [comment]
It occurs to me that it may be a good idea to measure and print the core-computation execution time for all tutorial programs.  This would make it easier for someone to play around with `CILK_NWORKERS` and observe parallel speedup.

Granted, Cilkscale and its API can serve the same purpose (and more), but I wonder if that would be "too complicated" for a few simple tutorial programs.

@neboat Any thoughts?

File: nqueens.c
Comment (neboat) [inline nqueens.c:107]: I'm getting a compile error due to this line:
`error: too few arguments to function call, expected 2, have 1`
```suggestion
  ctimer_print(t, "nqueens");
```

File: ctimer.h
Comment (neboat) [inline ctimer.h:425]: As a minor point, on one system where I'm testing this PR, `CLOCK_MONOTONIC` has only microsecond resolution, so all timings have three 0's at the end.  Timer resolution can vary between systems, and although you could use `clock_getres` to get the clock resolution and format the output, it might not be worth the effort.

**Review by neboat — CHANGES_REQUESTED** [review]
Tried out this PR and added some comments based on my tests.

File: ctimer.h
Comment (behoppe) [reply ctimer.h:425]: Thanks @neboat. 

@ailiop I think these requested changes await your approval? Or should I approve them? I'm not sure if I have clicked "resolve conversation" before...

Comment (ailiop) [reply ctimer.h:429]: @neboat Thank you, I merged your fix and added a note in the `ctimer.h` documentation regarding the resolution of the printed time.

I'm sorry this fell off my radar.

**Comment by ailiop** [comment]
@neboat @behoppe I think this PR is ready to merge when you get the chance.  (I don't have write access to the repo.)

File: ctimer.h
Comment (ailiop) [reply ctimer.h:429]: @behoppe Thanks for following up.  I'd lost track of this PR.

**Comment by behoppe** [comment]
Thanks @ailiop. Although I have "owner" privileges across the entire OpenCilk GitHub organization, I don't think I'm really the right person to be adjudicating PRs on the OpenCilk/tutorial repository.

@neboat what do you think? Shall I add @ailiop as an owner of the tutorial repository so that he can merge this PR and approve future tutorial PRs?

Screenshot 2023-02-15 at 13-44-38 Add timer utility and Cilkscale API example by ailiop · Pull Request #1 · OpenCilk_tutorial

**Comment by neboat** [comment]
@behoppe It is not necessary, nor appropriate, to grant someone full ownership of a repository just to allow them to merge PRs or approve future PRs.  I do support giving @ailiop write permissions to this repository in general.

For PRs, it is good practice to get additional verification that proposed changes do in fact work, i.e., on more than just the author's system.  For this PR, I would like to verify these changes on my own system before approving and merging the PR.  I will need some time to do that.

**Review by neboat — APPROVED** [review]

**Pull request merged**
